zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/CallActivityTest.java

```diff
@@ -813,6 +813,47 @@ public void shouldPropagateTreePathPropertiesWithIncident() {
         .hasOnlyCallingElementPath(ca1Index, ca2Index);
   }
 
+  @Test
+  public void shouldPropagateCorrectIndexesInCallingElementPathWhenMultipleProcessesInSameFile() {
+    // given
+    final String rootProcessId = "root-process";
+    // call activities { "call-parent", "call-child" } when sorted wll be {"call-child",
+    // "call-parent"}
+    final var ca1Index = 1;
+    final var ca2Index = 0;
+
+    ENGINE.deployment().withXmlClasspathResource("/processes/callActivity.bpmn").deploy();
```

Comment:
Generally, we prefer using the BPMN model API, as it highlights in the test what the process looks like. Navigation to the file is necessary to understand that the special thing about this test case is the order of the call activities in this process.

🔧 I suggest relying on the BPMN model API and an assertion to ensure the order of the call activities in the BPMN file.
🔧 If you don't want to rely on the model API, I suggest renaming the process file to clarify this special trait.

Reply:
In this test, I'm trying to test the behavior when we have multiple process deployed in the same bpmn resource (using collaboration lanes).

I couldn't find a way to build a similar model using the BPMN model API. But if there's a way to do it that I'm unaware of, I'd appreciate any help.